espresso/core/java/androidx/test/espresso/Espresso.java

```diff
@@ -183,6 +185,27 @@ public static void setFailureHandler(FailureHandler failureHandler) {
     BASE.failureHolder().update(checkNotNull(failureHandler));
   }
 
+  /**
+   * Returns the {@link UpdatableViewHierarchyRenderer} used by Espresso to render the view
+   * hierarchy as a string when an error happens.
+   *
+   * You can change the default {@link ViewHierarchyRenderer} implementation by calling
+   * {@link UpdatableViewHierarchyRenderer#updateViewHierarchyRenderer(ViewHierarchyRenderer)}.
+   *
+   * The default {@link ViewHierarchyRenderer} implementation can also be provided by 3rd party
+   * libraries by including a META-INF/services/androidx.test.espresso.ViewHierarchyRenderer
+   * file in their jar.
+   *
+   * If there are more than one library defining a
+   * META-INF/services/androidx.test.espresso.ViewHierarchyRenderer file in the classpath then
+   * Espresso will throw on errors. If that happens, you can call
+   * {@link UpdatableViewHierarchyRenderer#updateViewHierarchyRenderer(ViewHierarchyRenderer)}
+   * which takes precedence.
```

Comment:
how do yall feel about this?

Reply:
This does not seem like a good API to expose to users. It's mainly for our internal implementations, not something that users would need.

Reply:
Okay, I see where you are coming from. In the HumanReadables, `getViewHierarchyErrorMessage` method gives the view hierarchy info and the error messages. However, I believe we might want to leave the "problematic" views out of the scope of this view render.

Reply:
Libraries that support rendering can easily prevent this: make the Espresso adapter for the specific library a separate artifact, that way the library itself can be on the classpath the but the users have a choice of which adapter to put on the classpath for espresso to service load.

Reply:
SG, but can you clarify what should be done instead? Are you suggesting to simply remove this javadoc?

ie if using the service loader we have to decide what to do when there are multiple services available. Current approach is to throw, and I just added documentation explain what to do if that happens. Would you instead allow multiple services?